lightning/src/ln/channelmanager.rs

```diff
@@ -8840,6 +8901,11 @@ where
 			channel_manager.fail_htlc_backwards_internal(&source, &payment_hash, &reason, receiver);
 		}
 
+		for (source, preimage, downstream_value, downstream_closed, downstream_chan_id) in pending_claims_to_replay {
+			channel_manager.claim_funds_internal(source, preimage, Some(downstream_value),
```

Comment:
`downstream_closed` == the htlc was claimed on-chain? Not obvious to me why that's necessarily the case, but maybe it's our best guess?